src/Features/VisualBasic/Completion/KeywordRecommenders/Statements/ElseKeywordRecommender.vb

```diff
@@ -45,5 +45,23 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Completion.KeywordRecommenders.Stat
 
             Return SpecializedCollections.EmptyEnumerable(Of RecommendedKeyword)()
         End Function
+
+        Private Function IsNotDirectlyInElseBlock(context As VisualBasicSyntaxContext) As Boolean
+            Dim parent = context.TargetToken.Parent
+            Do While parent IsNot Nothing
+                If parent.IsKind(SyntaxKind.MultiLineIfBlock, SyntaxKind.ElseIfBlock) Then
+                    Return True
+                End If
+
+                If parent.IsKind(SyntaxKind.ElseBlock) Then
+                    Return False
+                End If
+
+                parent = parent.Parent
```

Comment:
Is there a case where we actually have to go up multiple levels, or should this just look at `parent` and then stop?